jobrunner/agent/main.py

```diff
@@ -74,11 +75,12 @@ def handle_single_task(task, api):
         except Exception as exc:
             span.set_status(trace.Status(trace.StatusCode.ERROR, str(exc)))
             span.record_exception(exc)
-            mark_task_as_error(
-                api,
-                task,
-                sys.exc_info(),
-            )
+            if not is_transient_error(exc):
```

Comment:
Is there a way we could tag the current span with an attribute to indicate non-fatal platform error?

Currently, we alert from INTERNAL_ERROR - db locked errors will rightly no longer trigger INTERNAL_ERROR with this change, so it would be good to beable to easily filter for such errors and alert on them?

It may be that setting the status to ERROR, is enough perhaps.  But maybe we want to tag db locked errors with a `db_locked=true` attribute, for easy finding?

Reply:
So, thinking about it, we could potentially have enough already.

We're marking the span as having status error, with the str(exc) as the message.

In theory, that should enough for us to filter on it and check for "database locked" in the error message in HC.

But it's a bit more involved, so another way would bet to get the current span get_current_span() inside is_transient_error, and add two attributes

1. transient_error=True (for all future transient errors)
2. db_locked=True (for specifically this error)

That will make it trivial for us to filter in HC.

Adding extra attributes are free, we should generally be bullish on adding them, I think.

Reply:
Would it work to change `is_transient_error()` to `get_transient_error_type()` (returning `None` for non-transient errors) and then annotate the span inside error handler? That would mean having something like `transient_error_type=db_locked` rather than `db_locked=True`, but I'm assuming that's OK from the Honeycomb point-of-view.

Reply:
yeah, that's better